docs/lang/articles/contribution/dev_install.md

```diff
@@ -4,318 +4,467 @@ sidebar_position: 2
-  up to 5 minutes to run all tests.
+- The `develop` command does not actually install anything but only symbolically links the source code to the deployment directory. 
+- The `install` command deep copies the source code so that end users need to rerun the command every time they modify the source code. 
 
-4. Execute `python3 setup.py clean` to clean up the local information of your
-  previous builds. This allows a fresh build without any cache from the previous
-  builds. Note that to uninstall the Taichi package from your Python
-  environment, please use `pip uninstall taichi`.
+The `develop` command serves the developers' needs better because edits to the Python files take effect immediately without the need to rerun the command. A rerun is needed only if you have modified the project's C extension or compiled files. See the [Development Mode](https://setuptools.pypa.io/en/stable/userguide/development_mode.html) for more information.
 
-## Conda
+:::
 
-To avoid directly installing Taichi's dependencies into your existing
-Python environment, we have provided a pre-defined `conda` environment.
-You can find the instructions [here](https://github.com/taichi-dev/taichi/blob/master/conda/README.md).
+</TabItem>
 
-:::note
-This step only helps you setup the development environment,
-you would still need to run `python3 setup.py develop` to re-build
-Taichi.
-:::
+<TabItem value="windows">
 
-## Docker
+1. Run powershell as administrator and `Set-ExecutionPolicy RemoteSigned`.
+2. Run [this script](https://github.com/taichi-dev/taichi_assets/releases/download/windows_scripts/install_deps.ps1) to install all necessary dependencies:
 
-For those who prefer to use Docker, we also provide a Dockerfile which
-helps setup the Taichi development environment with CUDA support based
-on Ubuntu docker image.
+   ```shell
+   .\install_deps.ps1
+   ```
+3. Reboot your computer. 
+4. Run powershell as administrator and run [this script](https://github.com/taichi-dev/taichi_assets/releases/download/windows_scripts/build_taichi.ps1) to build Taichi from souce: 
 
-:::note
-In order to follow the instructions in this section, please make sure
-you have the [Docker Desktop (or Engine for
-Linux)](https://www.docker.com/products/docker-desktop) installed and
-set up properly.
-:::
+   ```shell
+   .\build_taichi.ps1
+   ```
 
-### Build the Docker image
+</TabItem>
 
-From within the root directory of the taichi Git repository, execute
-`docker build -t taichi:latest .` to build a Docker image based off the
-local master branch tagged with _latest_. Since this builds the image
-from source, please expect up to 40 mins build time if you don't have
-cached Docker image layers.
+</Tabs>
 
-:::note
 
-In order to save the time on building Docker images, you could always
-visit our [Docker Hub
-repository](https://hub.docker.com/r/taichidev/taichi) and pull the
-versions of pre-built images you would like to use.
+## Install optional dependencies
```

Comment:
Should the optional dependencies be installed before compiling Taichi, or after that?

Reply:
These optional dependencies can be installed *either* before or after building Taichi. Sequence does not make much difference. Set me straight if I were wrong. @ailzhang

Reply:
CMake might need to detect such depedencies before it could enable the flags for related features.